### PR TITLE
ZOO-1687: update coding standard

### PR DESCRIPTION
**Pull request by mrmoree**
* update package-lock
* fix version number for stylelint-config-recommended
* stylelint config standard to 18.3.0
* add stylelint-config-rational-order and stylelint-order
* REVERTING from stylelint 10 to 9
* update eslint and eslint-config-standard
* add plugin standard
* es6 functions for BadCode.js
* Systemtest for eslint
* Ignoring stupid eslint 'Something went wrong! ' behavior

Zum testen in Zooroyal master einfach `composer require --dev "zooroyal/coding-standard" "dev-ZOO-1687"`und ausprobieren. (Danach composer.lock resetten)

### CODE REVIEW

File: tstest.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+function tstest(name = "Test"): string {
```
Comment (rdss-sknott) [inline tstest.ts]: Der File muss noch weg bevor gemerged werden kann

**Comment by rdss-sknott** [comment]
Beim ausführen von `coding-standard sca:all `erscheint fieser Fehler:
```
/var/www/zooroyal/htdocs/vendor/zooroyal/coding-standard/node_modules/micromatch/index.js:44
    let isMatch = picomatch(String(patterns[i]), { ...options, onResult }, true);
                                                   ^^^

SyntaxError: Unexpected token ...
    at createScript (vm.js:56:10)
    at Object.runInThisContext (vm.js:97:10)
    at Module._compile (module.js:549:28)
    at Object.Module._extensions..js (module.js:586:10)
    at Module.load (module.js:494:32)
    at tryModuleLoad (module.js:453:12)
    at Function.Module._load (module.js:445:3)
    at Module.require (module.js:504:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/var/www/zooroyal/htdocs/vendor/zooroyal/coding-standard/node_modules/stylelint/lib/isPathIgnored.js:4:20)
```

**Comment by jp-uq** [comment]
> Beim ausführen von `coding-standard sca:all `erscheint fieser Fehler:
> 
> ```
> /var/www/zooroyal/htdocs/vendor/zooroyal/coding-standard/node_modules/micromatch/index.js:44
>     let isMatch = picomatch(String(patterns[i]), { ...options, onResult }, true);
>                                                    ^^^
> 
> SyntaxError: Unexpected token ...
>     at createScript (vm.js:56:10)
>     at Object.runInThisContext (vm.js:97:10)
>     at Module._compile (module.js:549:28)
>     at Object.Module._extensions..js (module.js:586:10)
>     at Module.load (module.js:494:32)
>     at tryModuleLoad (module.js:453:12)
>     at Function.Module._load (module.js:445:3)
>     at Module.require (module.js:504:17)
>     at require (internal/module.js:20:19)
>     at Object.<anonymous> (/var/www/zooroyal/htdocs/vendor/zooroyal/coding-standard/node_modules/stylelint/lib/isPathIgnored.js:4:20)
> ```

der Fehler stammt aus dem styelint Befehlt. Bei dem eslint command kommt auch ein fehler hoch:
 Parsing error: File 'config/typescript/tsconfig.json' not found

**Comment by mrmoree** [comment]
> > Beim ausführen von `coding-standard sca:all `erscheint fieser Fehler:
> > ```
> > /var/www/zooroyal/htdocs/vendor/zooroyal/coding-standard/node_modules/micromatch/index.js:44
> >     let isMatch = picomatch(String(patterns[i]), { ...options, onResult }, true);
> >                                                    ^^^
> > 
> > SyntaxError: Unexpected token ...
> >     at createScript (vm.js:56:10)
> >     at Object.runInThisContext (vm.js:97:10)
> >     at Module._compile (module.js:549:28)
> >     at Object.Module._extensions..js (module.js:586:10)
> >     at Module.load (module.js:494:32)
> >     at tryModuleLoad (module.js:453:12)
> >     at Function.Module._load (module.js:445:3)
> >     at Module.require (module.js:504:17)
> >     at require (internal/module.js:20:19)
> >     at Object.<anonymous> (/var/www/zooroyal/htdocs/vendor/zooroyal/coding-standard/node_modules/stylelint/lib/isPathIgnored.js:4:20)
> > ```
> 
> der Fehler stammt aus dem styelint Befehlt. Bei dem eslint command kommt auch ein fehler hoch:
> Parsing error: File 'config/typescript/tsconfig.json' not found

es lint gucke ich mir an

**Comment by easybi-ck** [comment]
The bug is fixed!

```
/var/www/zooroyal/htdocs/assets/src/js/detail.js
  1:1  error  Definition for rule 'no-useless-catch' was not found  no-useless-catch

✖ 1 problem (1 error, 0 warnings)
```

**Pull request merged**